File: GameFramePro/Assets/Scripts/Core/Base/Module/NetWork/Download/Downloader/UnityWebRequest/AssetBundleDownloadManager.cs
using System;
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using UnityEngine.Networking;

namespace GameFramePro.NetWorkEx
{
    /// <summary>
    /// 下载AssetBundle 的下载器
    /// </summary>
    public class AssetBundleDownloadManager : UnityWebRequestDownLoadManager
    {

        protected static AssetBundleDownloadManager s_Instance = null;
        public static AssetBundleDownloadManager S_Instance
        {
            get
            {
                if (s_Instance == null)
                    s_Instance = new AssetBundleDownloadManager();
                return s_Instance;
            }
        }// 
  


        protected override UnityWebRequestDownloadTask GetDownloadTaskInstance(string taskUrl, Action<UnityWebRequest, bool, string> callback, UnityTaskPriorityEnum priorityEnum, params object[] otherParameter)
        {
            var newTask = mUnityWebRequestTaskPoolManager.GetItemFromPool();
            UnityWebRequest webRequest = new UnityWebRequest(taskUrl);
            webRequest.downloadHandler = new DownloadHandlerAssetBundle(taskUrl, uint.MaxValue);
            newTask.InitialedDownloadTask(taskUrl, webRequest, priorityEnum, callback);
            return newTask;
        }


        
    }
}

File: GameFramePro/Assets/.lpproj/Apis/Unity/CS/GameFramePro/NetWorkEx/AssetBundleDownloadManager.lua
---@module CS.GameFramePro.NetWorkEx
CS.GameFramePro.NetWorkEx = {}

---@class CS.GameFramePro.NetWorkEx.AssetBundleDownloadManager : CS.GameFramePro.NetWorkEx.UnityWebRequestDownLoadManager
CS.GameFramePro.NetWorkEx.AssetBundleDownloadManager = {}

---@property readonly CS.GameFramePro.NetWorkEx.AssetBundleDownloadManager.S_Instance : CS.GameFramePro.NetWorkEx.AssetBundleDownloadManager
CS.GameFramePro.NetWorkEx.AssetBundleDownloadManager.S_Instance = nil

---@return CS.GameFramePro.NetWorkEx.AssetBundleDownloadManager
function CS.GameFramePro.NetWorkEx.AssetBundleDownloadManager()
end

---@param taskUrl : CS.System.String
---@param crc : CS.System.UInt32
---@param callback : CS.System.Action
---@param priorityEnum : CS.GameFramePro.NetWorkEx.UnityTaskPriorityEnum
function CS.GameFramePro.NetWorkEx.AssetBundleDownloadManager:GetDataFromUrl(taskUrl, crc, callback, priorityEnum)
end